jobs/buildhub/scrape_archives.py

```diff
@@ -112,9 +89,25 @@ def archive(product, version, platform, locale, url, size, date, metadata=None):
             "mimetype": guess_mimetype(url),
             "size": size,
             "date": date,
-        },
-        "systemaddons": None
+        }
     }
+
+    if metadata:
+        # Example of metadata:
+        #  https://archive.mozilla.org/pub/thunderbird/candidates \
+        #  /50.0b1-candidates/build2/linux-i686/en-US/thunderbird-50.0b1.json
+        record['source']['revision'] = metadata["moz_source_stamp"]
+        channel = metadata.get("moz_update_channel", channel)
+        repository = metadata["moz_source_repo"].replace("MOZ_SOURCE_REPO=", "")
+        record['source']['repository'] = repository
+        record['source']['tree'] = repository.split("hg.mozilla.org/", 1)[-1]
+        buildid = metadata["buildid"]
+        builddate = datetime.datetime.strptime(buildid[:12], "%Y%m%d%H%M").isoformat()
+        record['build'] = {
+            "id": buildid,
+            "date": builddate,
+        }
```

Comment:
This is a micronit, but could you group together the obtention of metadata values and appart the assignations into `record`?